src/traces/bar/defaults.js

```diff
@@ -61,16 +60,19 @@ function supplyDefaults(traceIn, traceOut, defaultColor, layout) {
 function crossTraceDefaults(fullData, fullLayout) {
     var traceIn, traceOut;
 
-    function coerce(attr) {
-        return Lib.coerce(traceOut._input, traceOut, attributes, attr);
+    function coerce(attr, dflt) {
+        return Lib.coerce(traceOut._input, traceOut, attributes, attr, dflt);
     }
 
-    if(fullLayout.barmode === 'group') {
-        for(var i = 0; i < fullData.length; i++) {
-            traceOut = fullData[i];
+    for(var i = 0; i < fullData.length; i++) {
+        traceOut = fullData[i];
 
-            if(traceOut.type === 'bar') {
-                traceIn = traceOut._input;
+        if(traceOut.type === 'bar') {
+            traceIn = traceOut._input;
+            // This needs to happen here rather than in handleStyleDefaults() because
+            // it needs to happen after `layout.barcornerradius` has been coerced
+            coerce('marker.cornerradius', fullLayout.barcornerradius);
```

Comment:
Let's validate inputs here. Something like:
```
            var r = coerce('marker.cornerradius', fullLayout.barcornerradius);
            var validR = false;
            if(isNumeric(r)) {
              r = +r;
              if(r >= 0) validR = true;
            } else if(typeof r === 'string' && r.slice(-1) === '%') {
                r = +r.slice(0, -1);
                if(r >= 0 && r <= 50) {
                  r += '%';
                  validR = true;
                }
            }
            traceOut.marker.cornerradius = validR ? r : undefined;
```

We need similar logic for histogram as well as `barcornerradius`.

Reply:
Looks good -- I think there should be no upper bound on the percentage though; it will be limited to 50% at the plotting step anyway.

Reply:
@alexcjohnson what do you think here?

Reply:
Sounds reasonable to not put an upper bound - say someone uses a calculation for this, and the calculation accidentally goes over 50% (whether to 50.1% or to 200%), I think it'd be friendlier to result in maximum rounding rather than discard it and give no rounding.